bids/layout/layout.py

```diff
@@ -563,18 +566,24 @@ def index_file(self, f, overwrite=False):
                 an entry already exists.
         """
         if isinstance(f, six.string_types):
-            f = self.layout.get_file(f)
+            filename = self.layout.get_file(f).path
+        elif isinstance(f, tuple):
```

Comment:
What's the motivation for this? For methods like this, I'd prefer to minimize the number of valid types that can be passed as input. It's cleaner if the caller has to first extract a string or `BIDSFile`, I think. Plus, long term, I'd like to get rid of the namedtuple entirely—I think it mostly just adds confusion.

Reply:
Motivation was pointed to below this line: http://github.com/grabbles/grabbit/pull/84 .  In the effort of using fitlins with current pybids we ended up here with the `File` tuple which has no `.path` (see that issue for our past discussion).  Since our attempts are largely shots in the darkness trying to maneuver between fluid APIs, I did not want to try to change more than minimally necessary and this change was such a minimal change to stay compatible possibly with older and newer pybids API.

Reply:
I see no problems with it, as long as its backwards compatible. It does seem a bit redundant and confusing two have two types of objects that can be returned.

Reply:
I've opened #281 to deal with this.